### PR TITLE
Implement ranges::merge

### PR DESCRIPTION
**Pull request by CaseyCarter**
Partially addresses #39.

### CODE REVIEW

**Comment by miscco** [comment]
Looks great! *merge* it

File: stl/inc/algorithm [hunk 1]
```diff
@@ -6807,6 +6807,90 @@ _FwdIt3 merge(_ExPo&&, _FwdIt1 _First1, _FwdIt1 _Last1, _FwdIt2 _First2, _FwdIt2
...
+                _Merge_unchecked(_Get_unwrapped(_STD move(_First1)), _Uend(_Range1), _Get_unwrapped(_STD move(_First2)),
+                    _Uend(_Range2), _STD move(_Result), _Pass_fn(_Pred), _Pass_fn(_Proj1), _Pass_fn(_Proj2));
+            _Seek_wrapped(_First1, _STD move(_UResult.in1));
+            _Seek_wrapped(_First2, _STD move(_UResult.in2));
+            return {_STD move(_First1), _STD move(_First2), _STD move(_UResult.out)};
+        }
+        // clang-format on
+    private:
+        template <class _It1, class _Se1, class _It2, class _Se2, class _Out, class _Pr, class _Pj1, class _Pj2>
+        static constexpr merge_result<_It1, _It2, _Out> _Merge_unchecked(_It1 _First1, const _Se1 _Last1, _It2 _First2,
```
Comment (StephanTLavavej) [inline stl/inc/algorithm:6858]: I believe you conventionally `_NODISCARD` the internal helper, even when the public interface must be discardable.

File: stl/inc/algorithm [hunk 2]
```diff
@@ -6807,6 +6807,90 @@ _FwdIt3 merge(_ExPo&&, _FwdIt1 _First1, _FwdIt1 _Last1, _FwdIt2 _First2, _FwdIt2
...
+            auto _First1 = _RANGES begin(_Range1);
+            auto _First2 = _RANGES begin(_Range2);
+            auto _UResult =
+                _Merge_unchecked(_Get_unwrapped(_STD move(_First1)), _Uend(_Range1), _Get_unwrapped(_STD move(_First2)),
+                    _Uend(_Range2), _STD move(_Result), _Pass_fn(_Pred), _Pass_fn(_Proj1), _Pass_fn(_Proj2));
+            _Seek_wrapped(_First1, _STD move(_UResult.in1));
+            _Seek_wrapped(_First2, _STD move(_UResult.in2));
+            return {_STD move(_First1), _STD move(_First2), _STD move(_UResult.out)};
+        }
+        // clang-format on
```
Comment (mnatsuhara) [inline stl/inc/algorithm:6855]: According to line 6845, I think clang-format is already back on :)

File: tests/std/tests/P0896R4_ranges_alg_merge/test.cpp [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    static constexpr P elements2[] = {{13, 0}, {14, 0}, {10, 2}, {11, 3}, {12, 3}};
+    static constexpr P expected[]  = {
+        {0, 10}, {0, 11}, {0, 12}, {13, 0}, {14, 0}, {1, 10}, {1, 11}, {10, 2}, {3, 10}, {11, 3}, {12, 3}};
+
+    template <ranges::input_range R1, ranges::input_range R2, weakly_incrementable O>
+    static constexpr void call() {
+        using ranges::merge, ranges::merge_result, ranges::end, ranges::equal, ranges::iterator_t, ranges::less,
+            ranges::size;
+
+        { // Validate range overload
```
Comment (mnatsuhara) [inline tests/std/tests/P0896R4_ranges_alg_merge/test.cpp:39]: We discussed in the [ranges::remove family](https://github.com/microsoft/STL/pull/1005#discussion_r453018420) PR the idea of testing for the number of comparisons when part of the algorithm is abstracted into another function.  We erred on the side of adding additional test coverage in that case, though I don't feel incredibly strongly about it here.  Just something to consider.

Comment (CaseyCarter) [reply tests/std/tests/P0896R4_ranges_alg_merge/test.cpp:45]: Counting comparator added!

**Comment by StephanTLavavej** [comment]
`ranges::merge` merged. :rotating_light: Warning! :rotating_light: Pun potential exceeds design capacity.

**Pull request merged**